mineral-ui/mineral-ui
# fix(select,textarea,textinput): Support component-level theme variables

### Description

- Update Select, TextArea, and TextInput to style a properly themed base component
- Update all other files using `mapComponentThemes` to use a consistent code style
- Update snapshots

### Motivation and context

Closes #742 

Specifically, by styling a non-themed base component (FauxControl, in the case of TextInput), we were not supporting any of the theme variables defined on that base component.

### Screenshots, videos, or demo, if appropriate


https://742-component-themes--mineral-ui.netlify.com/
(_Unfortunately, that link won't help much this time._)

### How to test


1. `cd mineral-ui && npm start`
2. Update Select/TextArea/TextInput examples to render a themed version, where the theme sets variables defined by the base component ([TextInput](https://github.com/mineral-ui/mineral-ui/blob/master/src/library/TextInput/TextInput.js#L62-L81) for Select; [FauxControl](https://github.com/mineral-ui/mineral-ui/blob/master/src/library/FauxControl/FauxControl.js#L41-L62), for TextArea & TextInput). Something like:

    ```jsx
    // /website/app/demos/TextInput/examples/uncontrolled.js

    import _TextInput from '../../../../../library/TextInput/';
    import { createThemedComponent } from '../../../../../library/themes';

    const TextInput = createThemedComponent(_TextInput, {
      TextInput_borderColor: 'rebeccapurple'
    });

    export default {
      // unchanged
    };
    ```

3. Check that your theme overrides took hold.

### Types of changes

- Bug fix (non-breaking change which fixes an issue)
- Other (provide details below)

Simple refactor for code style.

### Checklist




### How does this PR make you feel?

![Oopsie](https://media.giphy.com/media/xT9IghUVWeHfJTP5Li/giphy.gif)
